base/reflection.jl

```diff
@@ -179,6 +194,10 @@ function methods(f::ANY,t::ANY)
     Any[m[3] for m in _methods(f,t,-1)]
 end
 function _methods(f::ANY,t::ANY,lim)
+    #special case Expr to allow methods(@time,Tuple{Any})
+    if _is_empty_macrocall(f)
+        f = _get_macro(current_module(),f.args[1])
```

Comment:
You can't do this in `_methods`. `_methods(f, ...)` must return the list of methods you could call by doing `f(...)` *with the specific object* `f`. The following (current) behavior is correct:

```
julia> e = Expr(:macrocall, symbol("@time"))
:(@time)

julia> e()
ERROR: MethodError: objects of type Expr are not callable
 in eval(::Module, ::Any) at ./boot.jl:275

julia> methods(e)
# 0 methods for generic function "(::Expr)":
```